py_wtf/cli/__init__.py

```diff
@@ -145,25 +150,29 @@ async def generate_test_index(dir: str | None) -> None:
     for proj_dir in fixtures.iterdir():
         if not (proj_json := proj_dir / "project.json").exists():
             continue
+        proj_name = ProjectName(proj_dir.name)
         proj_metadata = json.loads(proj_json.read_bytes())
         proj_info = ProjectMetadata(
+            proj_name,
             proj_metadata["version"],
             classifiers=proj_metadata.get("classifiers"),
             home_page=proj_metadata.get("home_page"),
             license=proj_metadata.get("license"),
             documentation_url=proj_metadata.get("documentation_url"),
-            dependencies=proj_metadata["dependencies"],
+            dependencies=parse_deps(proj_metadata["dependencies"]),
             summary=proj_metadata.get("summary"),
         )
         symbol_table = SymbolTable({FQName("alpha.bar"): ProjectName("project-alpha")})
         mods = [mod async for mod in index_dir(proj_dir, symbol_table)]
         proj = Project(
-            ProjectName(proj_dir.name),
+            proj_name,
             metadata=proj_info,
             modules=mods,
             documentation=[Documentation(proj_info.summary or "")],
         )
         (out_dir / f"{proj.name}.json").write_text(converter.dumps(_sort(proj)))
+    repo = ProjectRepository(out_dir)
+    repo.write_index(1662219954)
 
 
```

Comment:
`time.time()`? :D

Reply:
no it has to be deterministic :)

Reply:
Not obvious to me on 15 seconds of thinking why. Why? And also, why is a hardcoded timestamp the right (or even a good) "hash"?

Reply:
Because there's a pre-commit hook and CI check to make sure the test index is always up-to-date